core/src/test/java/io/grpc/internal/TestUtils.java

```diff
@@ -36,7 +41,41 @@
 /**
  * Common utility methods for tests.
  */
-final class TestUtils {
+public final class TestUtils {
+
+  /** Base class for a standard LoadBalancerProvider implementation. */
+  public abstract static class StandardLoadBalancerProvider extends LoadBalancerProvider {
```

Comment:
A fake `LoadBalancerProvider` is trivial to implement, although it might be a little verbose to always implement `isAvailable()`, `getPriority()`, etc in unit test. If your test case is just trying to verify the behavior of looking up the registry correctly, the real default registry suffices.

Reply:
This class is not about which registry to use, it's about registering fake policy.

Reply:
My point is:

If you are just verifying your provider's behavior of looking up a policy in the registry, you can use the default registry and look up existing policies such as 'round_robin'. The behavior of looking up 'round_robin' isn't different from looking up 'foo'.

If you really have to register a fake policy into the registry, fully implementing all `LoadBalancerProvider`'s methods is still trivial, do not need to have this `StandardLoadBalancerProvider`.

Reply:
Fully implementing all LoadBalancerProvider's methods is trivial but is just repeating boilerplates that's not relevant to the test at all, it's distracting to read the test, the new base class does provide some benefit.

Reply:
Whatever...